### PR TITLE
Create lookup for OS build ID

### PR DESCRIPTION
**Pull request by ysolomchenko**
Fixes https://github.com/open-telemetry/semantic-conventions/issues/1318

## Changes

Create a look up for OS build ID

Please provide a brief description of the changes here.

Note: if the PR is touching an area that is not listed in the [existing areas](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/README.md), or the area does not have sufficient [domain experts coverage](https://github.com/open-telemetry/semantic-conventions/blob/main/.github/CODEOWNERS), the PR might be tagged as [experts needed](https://github.com/open-telemetry/semantic-conventions/labels/experts%20needed) and move slowly until experts are identified.

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [ ] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

File: docs/resource/os.md [hunk 1]
```diff
@@ -15,31 +15,46 @@ In case of virtualized environments, this is the operating system as it is obser
...
-| `dragonflybsd` | DragonFly BSD | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
-| `freebsd` | FreeBSD | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
-| `hpux` | HP-UX (Hewlett Packard Unix) | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
-| `linux` | Linux | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
-| `netbsd` | NetBSD | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
-| `openbsd` | OpenBSD | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
-| `solaris` | SunOS, Oracle Solaris | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
-| `windows` | Microsoft Windows | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
-| `z_os` | IBM z/OS | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| Value          | Description                          | Stability                                                        |
```
Comment (joaopgrassi) [inline docs/resource/os.md:28]: Please do not make manual changes to auto-generated content of markdown files. These are generated by Weaver based on contents of the yaml model files. Please read the contributing guide: https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#1-modify-the-yaml-model

File: docs/resource/os.md [hunk 2]
```diff
@@ -15,31 +15,46 @@ In case of virtualized environments, this is the operating system as it is obser
...
+|         |                                                                                                      | contents of `/proc/sys/kernel/osrelease`                                      |
+
+### Implementation Notes
+
+1. **Windows**: The build ID can be retrieved from the registry key `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion`, where the `CurrentBuildNumber` value holds the desired information.
+2. **MacOS**: On MacOS, the build ID is stored in the `ProductBuildVersion` field within the plist files located at `/System/Library/CoreServices/SystemVersion.plist` and `/System/Library/CoreServices/ServerVersion.plist`. The first file is typically used for regular MacOS installations, while the second might be used in server configurations.
+3. **Linux**: Linux distributions typically store the build ID in the `/etc/os-release` or `/usr/lib/os-release` files under the `BUILD_ID` field. If these files or the `BUILD_ID` field are not available, the `os.build_id` can be obtained from the `/proc/sys/kernel/osrelease` file as a fallback.
```
Comment (joaopgrassi) [inline docs/resource/os.md:57]: This recommendation should probably be part of the yaml file and rendered automatically. https://github.com/open-telemetry/semantic-conventions/blob/main/model/resource/os.yaml#L15

**Review by joaopgrassi — DISMISSED** [review]

File: docs/resource/os.md (hunk 2 above)
Comment (ysolomchenko) [reply docs/resource/os.md:57]: Could you provide the file with the same implementation? I can't find it in the resources host.

Comment (joaopgrassi) [reply docs/resource/os.md:57]: Look for `notes: ` in the yaml files, like this https://github.com/open-telemetry/semantic-conventions/blob/main/model/trace/database.yaml#L259

markdown syntax should work there as well.

Comment (ysolomchenko) [reply docs/resource/os.md]: @joaopgrassi fixed

File: model/resource/os.yaml [hunk 3]
```diff
@@ -13,3 +13,16 @@ groups:
       - ref: os.name
       - ref: os.version
       - ref: os.build_id
+        note: >
+          Resource detector SHOULD fetch values from following sources:
```
Comment (joaopgrassi) [inline model/resource/os.yaml:17]:
```suggestion
          `build_id` values SHOULD be obtained from the following sources:
```
I think we usually omit the component that is "fetching" the values. So I suggest something like this.

**Review by joaopgrassi — APPROVED** [review]

**Review by lmolkova — APPROVED** [review]

**Pull request merged**